src/write_http.c

```diff
@@ -91,6 +98,24 @@ typedef struct wh_callback_s wh_callback_t;
 static char **http_attrs;
 static size_t http_attrs_num;
 
+static size_t wh_curl_write_callback(char *ptr, size_t size, size_t nmemb,
+                                     void *userdata) {
+
+  wh_callback_t *cb = (wh_callback_t *)userdata;
+  int len = 0;
+
+  DEBUG("write_http plugin: curl callback writing %zu bytes of output.", nmemb);
+  if (nmemb > cb->response_buffer_size - 1)
+    len = cb->response_buffer_size - 1;
+  else
+    len = nmemb;
+
+  memcpy(cb->response_buffer, ptr, len);
+  cb->response_buffer[cb->response_buffer_size] = '\0';
```

Comment:
This produces a memory corruption as it writer the `\0` past the end of the allocated storage. This should probably be
```
cb->response_buffer[len] = '\0';
```